Merge `net8.0` into `main`

mattleibow opened the pull request:
### Description of Change

Updating the "SR 1" branch with "GA" code.

dustin-wojciechowski committed:
Make Entitlements in Templates for MacCatalyst more consistent between debug and release configs. (#17833)

* Updated integration test to check for app-sandbox and network.client entitlements in debug config as well for maui-blazor.

* Added app-sandbox and network.client entitlements to both debug and release entitlement plists for maui-blazor and maui-mobile templates for MacCatalyst. get-task-allow is commented out for debug for maui-mobile but is not for maui-blazor.

* Removed unnecessary space

* Merged debug and release entitlement plists, edition conditionals in csproj

* Removed unnecessary items from csproj

mattleibow committed:
Revert "Work around the dotnet tool restore issues" (#17983)

Eilon committed:
Update Blazor Hybrid template to match ASP.NET Core Blazor Web template (#17946)

Fixes #16797

Eilon committed:
Enable <Nullable> in .NET MAUI project templates (#18019)

Fixes #17987

Eilon committed:
Set Nullable=enable in Control Samples projects (#18020)

This was to see if anything particularly unexpected happens due to making the templates have Nullable=enable by looking to see what happens in relatively large/complex projects. The conclusion is that it all seems to be just fine.

rmarinho committed:
Revert new packages names for iOS (#17980)

jonathanpeppers committed:
[android] update AOT profile for .NET 8 GA (#18061)

To used the very latest builds, I updated locally:

    <!-- dotnet/installer -->
    -<MicrosoftDotnetSdkInternalPackageVersion>8.0.100-rtm.23506.1</MicrosoftDotnetSdkInternalPackageVersion>
    +<MicrosoftDotnetSdkInternalPackageVersion>8.0.100-rtm.23513.4</MicrosoftDotnetSdkInternalPackageVersion>
    <!-- dotnet/runtime -->
    -<MicrosoftNETCoreAppRefPackageVersion>8.0.0-rtm.23509.5</MicrosoftNETCoreAppRefPackageVersion>
    +<MicrosoftNETCoreAppRefPackageVersion>8.0.0-rtm.23513.17</MicrosoftNETCoreAppRefPackageVersion>
    <!-- xamarin-android -->
    -<MicrosoftAndroidSdkWindowsPackageVersion>34.0.12</MicrosoftAndroidSdkWindowsPackageVersion>
    +<MicrosoftAndroidSdkWindowsPackageVersion>34.0.25</MicrosoftAndroidSdkWindowsPackageVersion>

Before updating the profile, an average of 10 runs on a Pixel 5 of a
`dotnet new maui` app:

    Average(ms): 557.9
    Std Err(ms): 1.73493515728975
    Std Dev(ms): 5.48634668973808

After updating the profile:

    Average(ms): 551.3
    Std Err(ms): 2.82075639973867
    Std Dev(ms): 8.92001494767058

These are even better numbers than we got in 2d8cf282. It's possible we
slightly beat .NET 7 now, I'll test.

github-actions[bot] committed:
[ci] Fix api leves on tests (#18133)

Co-authored-by: Rui Marinho <me@ruimarinho.net>

jsuarezruiz committed:
[Windows] Fix wrong Layout from hidden CollectionView  (#17659)

* Fix wrong Layout from hidden CollectionView

* Applied some changes

StephaneDelcroix committed:
[X] connect the right method to the event handler (#18043)

- fixes #17950

StephaneDelcroix committed:
[iOS] Keep textcolor on characterspacing (#17926)

* add tests

* fix

emaf committed:
[WinUI] Fixes window blinking when resizing (#18155)

This bug was introduced by https://github.com/dotnet/maui/pull/17069,
but the underlying cause is a bug on WinUI that makes the window blink when
setting `NavigationView.IsPaneVisible = false` (https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/47131770).

After testing the samples attached on the issues the original commit fixed, this change does not seem to re-introduce those issues.

Fixes https://github.com/dotnet/maui/issues/17724

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.9 (#18157)

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8438 -> To Version 17.0.8445

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18171)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.52

Microsoft.iOS.Sdk
 From Version 17.0.8438 -> To Version 17.0.8446

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.52

Microsoft.tvOS.Sdk
 From Version 17.0.8438 -> To Version 17.0.8446

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.52

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8445 -> To Version 17.0.8446

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231019.52

Microsoft.macOS.Sdk
 From Version 14.0.8438 -> To Version 14.0.8446

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

rmarinho committed:
Update rtm versions (#18167)

mattleibow committed:
Disable the new code for now (#18198)

mattleibow committed:
[net8.0] Add permission for high-speed sensors (#18180)

StephaneDelcroix committed:
[X] revert #5611 (#17733)

As reported in #17461, OnPlatform simplification cause some issues
- while used as a Resource (#17461, unable to unit test for now)
- doesn't type convert. On nodes shouldn't be replaced by the Value, but
  an element node (<On Platform="..." Value="Red" /> should generate
<Color x:Key="foo">Red</Color>. right now it only generated "Red"

- fixes #17461

jknaudt21 committed:
[iOS] Fix downscaled images not occupying correct space (#17120)

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18191)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.16

Microsoft.iOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8447

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.16

Microsoft.tvOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8447

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.16

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8446 -> To Version 17.0.8447

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.16

Microsoft.macOS.Sdk
 From Version 14.0.8446 -> To Version 14.0.8447

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.17

Microsoft.iOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8448

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.17

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8446 -> To Version 17.0.8448

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.17

Microsoft.macOS.Sdk
 From Version 14.0.8446 -> To Version 14.0.8448

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.17

Microsoft.tvOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8448

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.28

Microsoft.iOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8450

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.28

Microsoft.tvOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8450

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.28

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8446 -> To Version 17.0.8450

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.28

Microsoft.macOS.Sdk
 From Version 14.0.8446 -> To Version 14.0.8450

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.29

Microsoft.tvOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8451

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.29

Microsoft.macOS.Sdk
 From Version 14.0.8446 -> To Version 14.0.8451

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.29

Microsoft.iOS.Sdk
 From Version 17.0.8446 -> To Version 17.0.8451

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231020.29

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8446 -> To Version 17.0.8451

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/aspnetcore build 20231020.10 (#18224)

Microsoft.AspNetCore.Authentication.Facebook , Microsoft.AspNetCore.Authentication.Google , Microsoft.AspNetCore.Authentication.MicrosoftAccount , Microsoft.AspNetCore.Authorization , Microsoft.AspNetCore.Components , Microsoft.AspNetCore.Components.Analyzers , Microsoft.AspNetCore.Components.Forms , Microsoft.AspNetCore.Components.Web , Microsoft.AspNetCore.Components.WebView , Microsoft.AspNetCore.Metadata , Microsoft.JSInterop
 From Version 8.0.0-rtm.23518.6 -> To Version 8.0.0-rtm.23520.10

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

mattleibow committed:
Rework the Label mappers for FormattedText (#18105)

The current implementation did not recreate the FormattedText spans when certain properties changed: Font and TextColor

StephaneDelcroix committed:
[C] test for #17776 (#17796)

rmarinho committed:
Update net8 rtm versions (#18247)

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18248)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.1

Microsoft.tvOS.Sdk
 From Version 17.0.8451 -> To Version 17.0.8452

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.1

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8451 -> To Version 17.0.8452

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.1

Microsoft.macOS.Sdk
 From Version 14.0.8451 -> To Version 14.0.8452

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.1

Microsoft.iOS.Sdk
 From Version 17.0.8451 -> To Version 17.0.8452

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.9

Microsoft.iOS.Sdk
 From Version 17.0.8451 -> To Version 17.0.8453

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.9

Microsoft.tvOS.Sdk
 From Version 17.0.8451 -> To Version 17.0.8453

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.9

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8451 -> To Version 17.0.8453

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231023.9

Microsoft.macOS.Sdk
 From Version 14.0.8451 -> To Version 14.0.8453

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

mattleibow committed:
Merge remote-tracking branch 'origin/net8.0' into dev/merge-net8-into-main

# Conflicts:
#	.config/dotnet-tools.json
#	eng/Version.Details.xml
#	eng/Versions.props

rmarinho commented on build.ps1:
```diff
@@ -9,11 +9,7 @@ Param(
 )
 
 # Restore Cake tool
-# & dotnet tool restore
```
Main is not rtm so we  might need to revert this

mattleibow committed:
Add NuGet package information (#18197)

github-actions[bot] committed:
Correctly expand rows/columns when grid padding (#18276)

When expanding star definitions, the targetSize
parameter contains the padding (which is
removed). However, the calculation for the
the GridMinimum[Width|Height]() also has the
padding which also needs to be removed.

Co-authored-by: Matthew Leibowitz <mattleibow@live.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/aspnetcore build 20231023.8 (#18286)

Microsoft.AspNetCore.Authentication.Facebook , Microsoft.AspNetCore.Authentication.Google , Microsoft.AspNetCore.Authentication.MicrosoftAccount , Microsoft.AspNetCore.Authorization , Microsoft.AspNetCore.Components , Microsoft.AspNetCore.Components.Analyzers , Microsoft.AspNetCore.Components.Forms , Microsoft.AspNetCore.Components.Web , Microsoft.AspNetCore.Components.WebView , Microsoft.AspNetCore.Metadata , Microsoft.JSInterop
 From Version 8.0.0-rtm.23520.10 -> To Version 8.0.0-rtm.23523.8

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[net8.0] Update dependencies from xamarin/xamarin-macios (#18281)

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.3

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8454

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.3

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8454

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.3

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8454

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.3

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8454

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.5

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8455

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.5

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8455

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.5

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8455

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.5

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8455

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.19

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8456

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.19

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8456

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.19

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8456

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.19

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8456

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.20

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8457

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.20

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8457

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.20

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8457

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.20

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8457

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.21

Microsoft.iOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8458

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.21

Microsoft.tvOS.Sdk
 From Version 17.0.8453 -> To Version 17.0.8458

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.21

Microsoft.macOS.Sdk
 From Version 14.0.8453 -> To Version 14.0.8458

* Update dependencies from https://github.com/xamarin/xamarin-macios build 20231024.21

Microsoft.MacCatalyst.Sdk
 From Version 17.0.8453 -> To Version 17.0.8458

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

tj-devel709 committed:
Allow iOS Keyboard Scrolling to be turned off in Lifecycle Events (#18266)

* Add spacing

* Use a better implementation

PureWeen committed:
Setup legacy mappers (#18087)

* Route Controls Mapper to Core Mapper

* Update ControlsMapper.cs

* - fix test

mattleibow committed:
Merge remote-tracking branch 'origin/net8.0' into dev/merge-net8-into-main

# Conflicts:
#	eng/Version.Details.xml
#	eng/Versions.props
#	src/Core/src/Layouts/GridLayoutManager.cs
#	src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs
